dev/run-tests.py

```diff
@@ -0,0 +1,506 @@
+            print "[info] Detected changes in GraphX. Will run GraphX test suite."
+            test_suite.append("GRAPHX")
+
+        return set(test_suite)
+    else:
+        # we aren't in the Amplab environment so simply run all tests
+        test_suite.append("ALL")
+        return set(test_suite)
+
+
+def run_scala_tests_maven(test_profiles):
+    mvn_test_goals = ["test", "--fail-at-end"]
+    profiles_and_goals = test_profiles + mvn_test_goals
+
+    print "[info] Running Spark tests using Maven with these arguments:",
+    print " ".join(profiles_and_goals)
+
+    exec_maven(profiles_and_goals)
+
+
+def run_scala_tests_sbt(test_modules, test_profiles):
+    # declare the variable for reference
+    sbt_test_goals = None
+
+    if "ALL" in test_modules:
+        sbt_test_goals = ["test"]
+    else:
+        # if we only have changes in SQL build a custom test list
+        if "SQL" in test_modules and "CORE" not in test_modules:
+            sbt_test_goals = ["catalyst/test",
+                              "sql/test",
+                              "hive/test",
+                              "hive-thriftserver/test",
+                              "mllib/test",
+                              "examples/test"]
+        if "MLLIB" in test_modules and "CORE" not in test_modules:
+            sbt_test_goals = sbt_test_goals + ["mllib/test",
+                                               "examples/test"]
+        if "STREAMING" in test_modules and "CORE" not in test_modules:
+            sbt_test_goals = sbt_test_goals + ["streaming/test",
+                                               "streaming-flume/test",
+                                               "streaming-flume-sink/test",
+                                               "streaming-kafka/test",
+                                               "streaming-mqtt/test",
+                                               "streaming-twitter/test",
+                                               "streaming-zeromq/test",
+                                               "examples/test"]
+        if "GRAPHX" in test_modules and "CORE" not in test_modules:
+            sbt_test_goals = sbt_test_goals + ["graphx/test",
+                                               "examples/test"]
+        if not sbt_test_goals:
+            sbt_test_goals = ["test"]
+
+    profiles_and_goals = test_profiles + sbt_test_goals
+
+    print "[info] Running Spark tests using SBT with these arguments:",
+    print " ".join(profiles_and_goals)
+
+    exec_sbt(profiles_and_goals)
+
+
+def run_scala_tests(build_tool, hadoop_version, test_modules):
+    """Function to properly execute all tests passed in as a set from the
+    `determine_test_suites` function"""
+    set_title_and_block("Running Spark unit tests", "BLOCK_SPARK_UNIT_TESTS")
+
+    test_modules = set(test_modules)
+
+    # if the Spark SQL tests are enabled, run the tests with the Hive profiles
+    # enabled.
+    if "SQL" in test_modules:
```

Comment:
And then once we've done that, just get rid of the `if` -`else` and just say `test_profiles(hadoop_version, hive_profiles=hive_profiles)`